src/Substitute.ml

```diff
@@ -289,6 +281,7 @@ let rec statement_substitute (tsubst : T.TypeVarId.id -> T.ety)
       let p = place_substitute tsubst p in
       let rvalue = rvalue_substitute tsubst rvalue in
       A.Assign (p, rvalue)
+  | A.AssignGlobal g -> A.AssignGlobal g
```

Comment:
Add a comment explaining that globals shouldn't have type parameters (and sanity check?)